test/litmus_test.exs

```diff
@@ -12,21 +12,37 @@ defmodule LitmusTest do
         },
         "user" => %Litmus.Type.String{
           max_length: 6,
-          min_length: 3
+          min_length: 3,
+          regex: %Litmus.Type.String.Regex{
+            pattern: ~r/^[a-zA-Z0-9_]*$/
+          },
+          trim: true
         },
         "password" => %Litmus.Type.String{
           length: 4,
-          required: true
+          required: true,
+          trim: true
+        },
+        "zip_code" => %Litmus.Type.String{
+          min_length: 5,
+          regex: %Litmus.Type.String.Regex{
+            pattern: ~r/^\d{3,}(?:[-\s]?\d*)?$/,
+            error_message: "zip_code must be in a valid zip or zip+4 format"
+          },
+          trim: true
```

Comment:
Do we need this `zip_code` part of the schema?

Reply:
If not, I say let's remove it.

Reply:
agreed. we already have a string with a regex field. let's keep this simple